db/trie/kvstore.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2018 IoTeX
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package trie
+
+import (
+	"context"
+)
+
+// KVStore defines an interface for storing trie data as key-value pair
+type KVStore interface {
```

Comment:
why need to define a new KVStore interface? i think this is duplicate and might cause confusion
we cannot use existing KVStore with some tweak?

Reply:
not really. I think this is a good practice of using interface. You will find the differences of KVStore and the KVStore defined in DB.

Reply:
Using interface is of course good. My point is that we should re-use existing code/interface that has stood test of time, and avoid duplicating similar code if not necessary. I saw the interface is almost same except does not have 'bucket', that is why asking if we could re-use existing one with some tweak.

if you feel strongly a new interface is needed, can we name it differently like TrieKVStore or KVStoreNoBucket?

Reply:
The interface defined here is to help us isolate trie as an independent package.